File: src/main/java/com/n2/portal/service/ExpenseService.java
package com.n2.portal.service;

import com.n2.portal.dto.ExpenseDTO;
import com.n2.portal.model.expense.Expense;

import java.util.Date;
import java.util.List;

/**
 * Created by dhnhan on 20/10/2016.
 */
public interface ExpenseService {
    public Expense saveExpense(ExpenseDTO dto);

    public List<Expense> getExpenses(Date date,String gran, Long id);
}

File: src/main/webapp/n2/js/expense/services/expense.service.js
/**
 * Created by dhnhan on 21/10/2016.
 */
expenseModule.factory('ExpenseService', ['N2Service', 'METHOD', function (N2Service, METHOD) {
    return {
        urlEx: '/n2/api/expense',
        urlSpend: '/n2/api/spend',
        getExpense: function (year, month) {
            var params = {
                year: year,
                month: month
            };
            return N2Service.requestRest(this.urlEx, METHOD.GET, params);
        },
        saveExpense: function (name, date) {
            var params = {
                name: name,
                date: date
            };
            return N2Service.requestRest(this.urlSpend, METHOD.POST, params);
        },
        getExpenseOfSpend: function (date) {
            var params = {
                date: date
            };
            return N2Service.requestRest(this.urlSpend, METHOD.GET, params);
        },
        updateSpendCategory: function (id, name) {
            var params = {
                name: name
            };
            return N2Service.requestRest(this.urlSpend + '/' + id, METHOD.PUT, params);
        },
        deleteSpendCategory: function (id) {
            return N2Service.requestRest(this.urlSpend + '/' + id, METHOD.DELETE);
        }
    }
}]);